analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/LocksReleasedAllPaths.Monitor.Linear.cs

```diff
@@ -126,7 +127,42 @@ public void Method13(string arg)
     {
         Monitor.Exit(obj);
         Console.WriteLine(arg.Length);
-        Monitor.Enter(obj); // FN
+        Monitor.Enter(obj); // Noncompliant {{Unlock this lock along all executions paths of this method.}}
+    }
+
+    public void FieldReference_WithThis(string arg)
+    {
+        Monitor.Exit(this.obj);
+        Console.WriteLine(arg.Length);
+        Monitor.Enter(this.obj); // Noncompliant {{Unlock this lock along all executions paths of this method.}}
+    }
+
+    public void FieldReference_WithThis_Asimetric1(string arg)
```

Comment:
Typo
```suggestion
    public void FieldReference_WithThis_Asymmetric1(string arg)
```
or this, to avoid the problem
```suggestion
    public void FieldReference_WithThis_Mixed1(string arg)
```
It actually took me a while to realize what the asymmetry is. I'd prefer mixed.